RandLAPACK/misc/rl_util.hh

```diff
@@ -188,14 +188,10 @@ T cond_num_check(
     int64_t m,
     int64_t n,
     const T* A,
-    T* A_cpy,
-    T* s,
     bool verbose
 ) {
-
-    // TODO: GET RID OF THE INTERNAL ALLOCATIONS
-    A_cpy = ( T * ) calloc( m * n, sizeof( T ) );
-    s     = ( T * ) calloc( n, sizeof( T ) );
+    T* A_cpy = ( T * ) calloc( m * n, sizeof( T ) );
+    T* s     = ( T * ) calloc( n, sizeof( T ) );
```

Comment:
Personally I agree that we'd like to avoid the internal allocation. I suggest you allow workspace as an optional argument, a-la ...
```cpp
T cond_num_check(
    int64_t m,
    int64_t n,
    const T* A,
    bool verbose,
    T* work = nullptr,
    int64_t lwork = 0
) {
    int64_t allocate_more_work = lwork < m * n + n;
    if (allocate_more_work) {
        work = new T[m * n + n];
    }
    // use aliases for different parts of workspace
    T* A_cpy = work;
    T* s = work + m*n;

    // ---
    // --- do the work to compute output, "val" ---
    // ---

    if (allocate_more_work) {
        delete [] work;
    } 
    
    return val;
}
```
A slightly more efficient implementation would recognize when ``lwork >= m * n`` (big enough to hold ``A_cpy``) and yet ``lwork < m * n + n`` (not enough to hold both ``A_cpy`` and ``s``).

Reply:
Dealing with proper workspace logic is tedious and deserves a separate PR, especially when it comes to the old code.
For now, I wanted to make things consistent with the rest of the library.

Reply:
Fine with me. I'd just want to keep comments like ``// TODO: GET RID OF THE INTERNAL ALLOCATIONS`` in that case.